third_party/maya/lib/usdMaya/MayaMeshWriter.cpp

```diff
@@ -38,6 +38,49 @@ PXR_NAMESPACE_OPEN_SCOPE
 
 PXRUSDMAYA_REGISTER_ADAPTOR_SCHEMA(MFn::kMesh, UsdGeomMesh);
 
+namespace {
+
+void _exportReferenceMesh(UsdGeomMesh& primSchema, MObject obj) {
+    MStatus status = MS::kSuccess;
+    MFnDependencyNode dNode(obj, &status);
+    if (!status) { return; }
+
+    auto referencePlug = dNode.findPlug("referenceObject", &status);
+    if (!status || referencePlug.isNull()) { return; }
+
+    MPlugArray conns;
+    referencePlug.connectedTo(conns, true, false);
+    if (conns.length() == 0) { return; }
+
+    auto referenceObject = conns[0].node();
+    if (!referenceObject.hasFn(MFn::kMesh)) { return; }
+
+    MFnMesh referenceMesh(referenceObject, &status);
+    if (!status) { return; }
+
+    const auto* mayaRawPoints = referenceMesh.getRawPoints(&status);
+    const auto numVertices = referenceMesh.numVertices();
+    VtArray<GfVec3f> points(numVertices);
+    for (auto i = decltype(numVertices){0}; i < numVertices; ++i) {
```

Comment:
Also here -- if we explicitly declare the type for numVertices, we'd prefer to avoid the `decltype` and do `for (int i = 0; i < numVertices; ++i) ...`.